pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -392,7 +392,10 @@ private List<StartupParam> getParametersForStartup(String user, String database,
 
     if (assumeVersion.getVersionNum() >= ServerVersion.v9_0.getVersionNum()) {
       // User is explicitly telling us this is a 9.0+ server so set properties here:
-      paramList.add(new StartupParam("extra_float_digits", "3"));
+      if (assumeVersion.getVersionNum() < ServerVersion.v12.getVersionNum()) {
+        // extra_float_digits is meaningless in this case starting from v12  
```

Comment:
Be nice to add a link to https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-EXTRA-FLOAT-DIGITS in the comment